File: output/gemma4-e4b/avatar/Java/Python/atcoder_ABC132_A.py
from collections import Counter
import sys

def solve():
    # Read the input string (assuming it's a single token/word)
    try:
        s = sys.stdin.read().strip()
    except EOFError:
        s = ""

    if not s:
        print("No")
        return

    # Count character frequencies
    counts = Counter(s)

    # Condition 1: All characters must appear exactly twice
    all_counts_are_two = all(count == 2 for count in counts.values())

    # Condition 2: There must be exactly two distinct characters
    num_distinct_chars = len(counts)

    if all_counts_are_two and num_distinct_chars == 2:
        print("Yes")
    else:
        print("No")

solve()

File: output/gemma4-e4b/avatar/Python/Java/atcoder_ABC132_A.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class atcoder_ABC132_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String s = scanner.nextLine();
        scanner.close();

        // Use a HashMap to store character counts (equivalent to Python's Counter)
        Map<Character, Integer> counts = new HashMap<>();
        for (char c : s.toCharArray()) {
            counts.put(c, counts.getOrDefault(c, 0) + 1);
        }

        // Check if the number of unique characters (Sa) is exactly 2
        if (counts.size() != 2) {
            System.out.println("No");
            return;
        }

        // Check if every unique character appears exactly twice
        boolean allCountsAreTwo = true;
        for (int count : counts.values()) {
            if (count != 2) {
                allCountsAreTwo = false;
                break;
            }
        }

        if (allCountsAreTwo) {
            System.out.println("Yes");
        } else {
            System.out.println("No");
        }
    }
}